pandas/core/dtypes/cast.py

```diff
@@ -209,7 +209,10 @@ def changeit():
                 om = other[mask]
                 om_at = om.astype(result.dtype)
                 if (om == om_at).all():
-                    new_result = result.values.copy()
+                    if isinstance(result, np.ndarray):
+                        new_result = result.copy()
+                    else:
```

Comment:
what other types are here? a Series?

this should be converted at a higher level (at the beginning of the function)